examples/yjs/src/y-electric.ts

```diff
@@ -0,0 +1,495 @@
+> extends ObservableV2<YProvider> {
+  private doc: Y.Doc
+
+  private operations: {
+    options: OperationSteamOptions
+    endpoint: string
+  }
+
+  private awareness?: {
+    options: AwarenessSteamOptions
+    endpoint: string
+    protocol: awarenessProtocol.Awareness
+  }
+
+  private resumeStateProvider?: ResumeStateProvider
+  private resume: ResumeState
+
+  private _ready: boolean
+  private _connected: boolean
+  private _synced: boolean
+
+  private sendingAwarenessState: boolean = false
+  private pendingAwarenessUpdate: AwarenessUpdate | null = null
+
+  private operationsHandler: (update: Uint8Array, origin: unknown) => void
+  private awarenessHandler: (update: AwarenessUpdate, origin: unknown) => void
+  private unsubscribeShapes?: () => void
+  private exitHandler?: () => void
+
+  private onSendError?: (error: unknown, context: string) => Promise<boolean>
+
+  private fetchClient?: typeof fetch
+
+  // TODO: implement database provider for Electric
+  // - Keep document state in same place as shape resume state
+  // - Compute diff from local update state instead of persisting
+  //   the batch of pending changes
+
+  /**
+   * Creates a new ElectricProvider instance that connects YJS documents to Electric SQL.
+   *
+   * @constructor
+   * @param {Object} options - Configuration options for the provider
+   * @param {Y.Doc} options.doc - The YJS document to be synchronized
+   * @param {OperationsShapeOptions} options.operations.options - Options for the operations shape stream
+   * @param {string} options.operations.endpoint - URL endpoint for sending operations
+   * @param {AwarenessShapeOptions} options.awareness.options - Options for the awareness shape stream
+   * @param {string} options.awareness.endpoint - URL endpoint for sending awareness states
+   * @param {awarenessProtocol.Awareness} options.awareness.protocol - The awareness protocol implementation
+   * @param {ResumeStateProvider} [options.resumeStateProvider] - Alternatively, you can use a provider for loading/saving resume state
+   * @param {Observable<string>} [options.databaseProvider] - Observable for loading/saving document state (e.g. IndexeddbPersistence)
+   * @param {boolean} [options.connect=true] - Whether to automatically connect upon initialization
+   * @param {Function} [options.onSendError] - Error handler for sending operations/awareness changes
+   * @param {Function} [options.fetchClient] - Custom fetch implementation to use for sending operations/awareness changes
+   */
+  constructor({
+    doc,
+    operations,
+    awareness,
+    resumeStateProvider,
+    databaseProvider,
+    connect = true,
+    onSendError,
+    fetchClient,
+  }: {
+    doc: Y.Doc
+    operations: {
+      options: OperationsShapeOptions
+      endpoint: string
+    }
+    awareness?: {
+      options: AwarenessShapeOptions
+      endpoint: string
+      protocol: awarenessProtocol.Awareness
+    }
+    resumeStateProvider?: ResumeStateProvider
```

Comment:
I think these can be better named, `options` is a shape options and should be named as such. And make it clear the endpoint is the url that you sending the operation to.

In PGLite-sync we named the shape options param `shape` maybe flow the same pattern here?

For the endpoint url, maybe `sendUrl`? Under the shape options we use `url` rather than `endpoint` so i think it's better to use url for consistency.

```suggestion
    operations: {
      shape: OperationsShapeOptions
      sendUrl: string
    }
    awareness?: {
      shape: AwarenessShapeOptions
      sendUrl: string
      protocol: awarenessProtocol.Awareness
    }
```